### PR TITLE
Add support for OriginalDate tag

### LINKED ISSUE #244: Using original date instead of release date
This is a new iteration of issue #145, which has been closed some time ago. In the meantime, MusicPlayerDaemon/libmpdclient@e5d0e50 added `OriginalDate` support to libmpdclient. So from v2.12 onwards we have support for that tag.

### PR DESCRIPTION
**Pull request by Holzhaus**
This adds support for the OriginalDate tag. The OriginalDate is supported by MPD since MusicPlayerDaemon/libmpdclient@e5d0e50 which added OriginalDate file tag support to libmpdclient.

The actual filetype dependent tag names are taken from the "Original Release Date" entry at: https://picard.musicbrainz.org/docs/mappings/

This also adds the `Y` char for OriginalDate tag columns (as opposed to `y` for the normal Date tag).

This resolves #244.

*Note:* ~~I wasn't able to properly test this yet. As soon as I tested it and everything seems to work I will remove the [WIP] prefix.~~ Tested and apparently works fine.

### CODE REVIEW

**Comment by Holzhaus** [comment]
I tested it, works fine. It can be merged now ;-)

**Comment by arybczak** [comment]
Thanks. A few things:
- configure.ac needs to require libmpdclient 2.14, otherwise tag changes can't be observed within ncmpcpp (I'm running 2.13).
- Did you test the local filesystem browser? Editing there doesn't seem to change the tag for me, but it should even with older libmpdclient as then taglib is used to fetch tags.
- Please update example config file and man page with new format option.
- Do you think it makes sense to include that tag also in search engine?

File: src/settings.cpp [hunk 1]
```diff
@@ -436,6 +436,8 @@ bool Configuration::read(const std::vector<std::string> &config_paths, bool igno
 				return MPD_TAG_ALBUM_ARTIST;
 			else if (v == "date")
 				return MPD_TAG_DATE;
+			else if (v == "originaldate")
```
Comment (arybczak) [inline src/settings.cpp]: It's better to make it `original_date` for consistency as albumartist is `album_artist`.

**Comment by Holzhaus** [comment]
I made the necessary changes to `configure.ac`, the example config, the man-page and the search engine. I didn' t check the local filesystem browser (I am only running a remote instance), but apparently I forgot to add write support for the tag (Xiph/ID3v2).

File: src/settings.cpp (hunk 1 above)
Comment (Holzhaus) [reply src/settings.cpp]: done.

**Comment by arybczak** [comment]
The patch looks good, but there is one more issue to solve. It looks like I made a mistake with the libmpdclient version - it was added in 2.12, not 2.14, but according to https://github.com/MusicPlayerDaemon/MPD/blob/master/NEWS OriginalDate support is present only in MPD 0.21, which is not released yet. Hence for the majority of people who don't run this version this results in a confusion as you can write the tag and read it in a local browser, but you can't read it from MPD database.

I think the presence of this tag needs to be tied to the MPD version ncmpcpp is currently connected to.

**Comment by arybczak** [comment]
Ping.

**Comment by Holzhaus** [comment]
Sorry, I'm a bit busy right know. I'll try to implement it in a server version dependant way as soon as I have some spare time, maybe in a week to two.

**Comment by larson-carter** [comment]
@Holzhaus ping

**Comment by larson-carter** [comment]
I'm going to close this until the Author can come back and give me an update.

**Comment by mxmilkiib** [comment]
Might it be possible to look at this again? Ubuntu 20.04 LTS [has](https://repology.org/project/mpd/versions) 0.21.22 now.

**Pull request closed without merging**